app/models/metric/ci_mixin/capture.rb

```diff
@@ -33,25 +34,26 @@ def perf_capture(interval_name, start_time = nil, end_time = nil)
       return
     end
 
+    metrics_capture = perf_capture_object
     start_time, end_time = fix_capture_start_end_time(interval_name, start_time, end_time)
-    start_range, end_range, counters_data = just_perf_capture(interval_name, start_time, end_time)
+    start_range, end_range, counters_data = just_perf_capture(interval_name, start_time, end_time, metrics_capture)
```

Comment:
Was this to avoid calling `perf_capture_metrics` via the delegate?

Reply:
Ah I see you're using it more than once now and that's not memoized

Reply:
Actually the `perf_capture_object` will have multiple objects in it.
So we're explicitly building it with the targets and then using that object for our multi object metrics collection